integration/deltaspike-data/impl/src/main/java/com/blazebit/persistence/impl/deltaspikedata/impl/handler/EntityViewRepositoryHandler.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2014 - 2017 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.deltaspikedata.impl.handler;
+
+import com.balzebit.persistence.impl.deltaspikedata.api.EntityViewRepository;
+import com.blazebit.persistence.CriteriaBuilder;
+import com.blazebit.persistence.CriteriaBuilderFactory;
+import com.blazebit.persistence.PaginatedCriteriaBuilder;
+import com.blazebit.persistence.view.EntityViewManager;
+import com.blazebit.persistence.view.EntityViewSetting;
+import org.apache.deltaspike.data.impl.property.Property;
+import org.apache.deltaspike.data.impl.property.query.NamedPropertyCriteria;
+import org.apache.deltaspike.data.impl.property.query.PropertyQueries;
+
+import javax.inject.Inject;
+import javax.persistence.metamodel.SingularAttribute;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.apache.deltaspike.core.util.ArraysUtils.isEmpty;
+
+/**
+ * @author Moritz Becker
+ * @since 1.2.0
+ */
+public class EntityViewRepositoryHandler<E, V, PK extends Serializable>
+        implements EntityViewRepository<E, V, PK>, EntityViewDelegateQueryHandler {
+
+    @Inject
+    @EntityViewContext
+    private EntityViewCdiQueryInvocationContext context;
+
+    @Inject
+    private CriteriaBuilderFactory cbf;
+
+    @Inject
+    private EntityViewManager evm;
+
+    @Override
+    public V findBy(PK pk) {
+        CriteriaBuilder<E> cb = createCriteriaBuilder();
+        String idAttribute = evm.getMetamodel().view(viewClass()).getIdAttribute().getName();
```

Comment:
This is not correct, you have to access the EntityType of the ManagedViewType and look for it's id attribute.

Reply:
Why not use the `idAttribute()` method instead?